### PR TITLE
Fix 1sec timeouts and propagate errors

### PR DESCRIPTION
**Pull request by aetheon**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -46,66 +47,71 @@ idle.tick = function (callback) {
...
-idle.addListener = function (shouldSeconds, callback) {
+idle.addListener = function (intervalSec, callback) {
 	var isAfk = false;
 
-	var listenerId = listeners.push(true) - 1;
-	var timeoutRef = null;
+	var listenerIndex = listeners.push(true) - 1;
+    var defaultIntervalMsec = intervalSec * 1000;
+	var timeoutId = null;
+    var lastCheckDateMsec = null;
```
Comment (emmetcampion) [inline index.js]: Indentation seems to be off here and throughout the file

File: index.js [hunk 2]
```diff
@@ -46,66 +47,71 @@ idle.tick = function (callback) {
...
-			else {
-				timeoutRef = setTimeout(checkIsAway, s);
+        intervalDurationMsec = lastCheckDateMsec ? Date.now() - lastCheckDateMsec : Number.MAX_VALUE;
+        lastCheckDateMsec = Date.now();
+
+		idle.tick(function(idleSeconds, error){
+
+            if(error) {
+                callback({ id: listenerIndex }, error);
+                timeoutId = setTimeout(checkIsAway, nextIntervalDurationMsec);
```
Comment (emmetcampion) [inline index.js]: `nextIntervalDurationMsec` is not defined right?

File: index.js [hunk 3]
```diff
@@ -46,66 +47,71 @@ idle.tick = function (callback) {
...
-
-				isAfk = false;
-				timeoutRef = setTimeout(checkIsAway, whenSeconds * s);
-			}
-			else if (whenSeconds > 0 && !isAfk){
-				timeoutRef = setTimeout(checkIsAway, whenSeconds * s);
-			}
-			else {
-				timeoutRef = setTimeout(checkIsAway, s);
+        intervalDurationMsec = lastCheckDateMsec ? Date.now() - lastCheckDateMsec : Number.MAX_VALUE;
```
Comment (emmetcampion) [inline index.js]: Add a comment here

**Comment by aetheon** [comment]
@emmetcampion had to convert tabs to spaces :(

**Pull request merged**